lib/types.ts

```diff
@@ -0,0 +1,34 @@
+import { core } from '@balena/jellyfish-types';
+
+interface ContractDefinition<T>
+	extends Omit<core.ContractDefinition<T>, 'slug'> {}
+
+export interface TransformerContract
+	extends core.Contract<{
+		targetPlatform?: string;
+	}> {}
+
+// a general form of this would make sense in a "Transformers SDK"
+export type InputManifest<InputData> = {
```

Comment:
Most of the time when using contracts we use the whole contract interface instead of just the data hunk. Something like this might be more ergonomic.
```suggestion
export type InputManifest<T extends core.Contract = core.Contract> = {
```